packages/repository/src/decorators/relation.repository.decorator.ts

```diff
@@ -0,0 +1,61 @@
+import {Entity} from '../model';
+import {EntityCrudRepository} from '../repositories/repository';
+import {Class} from '../common-types';
+import {
+  HasManyDefinition,
+  hasManyRepositoryFactory,
+  RelationDefinitionBase,
+} from '../repositories/relation.factory';
+import {RelationType, RELATIONS_KEY} from './relation.decorator';
+import {
+  inject,
+  BindingKey,
+  Context,
+  Injection,
+  MetadataInspector,
+} from '@loopback/context';
+import {DefaultCrudRepository} from '../repositories/legacy-juggler-bridge';
+
+/**
+ * Decorator for hasMany
+ * @param targetRepo
+ * @returns {(target:any, key:string)}
+ */
+export function hasManyRepository<T extends Entity>(
+  targetRepo: Class<EntityCrudRepository<T, typeof Entity.prototype.id>>,
+) {
+  return function(target: Object, key: string) {
+    inject(
+      BindingKey.create<typeof targetRepo>(`repositories.${targetRepo.name}`),
+      {sourceRepoProto: target},
+      resolver,
+    )(target, key);
+  };
+
+  async function resolver(ctx: Context, injection: Injection) {
+    const tRepo = await ctx.get<
```

Comment:
What does `t` in `tRepo` stand for? Please use full names instead of abbrevitations.